src/AllHorses/AllHorses.js

```diff
@@ -1,33 +1,37 @@
-import React from "react";
+import React, { useState } from "react";
 import "./AllHorses.scss";
 import { Link } from 'react-router-dom';
 import mockData from "../mockData";
 import HorseCard from "../HorseCard/HorseCard";
+import { fetchAllHorses } from '../graphqlQueries.js';
+import { useQuery } from '@apollo/client';
 
 function AllHorses() {
-  const allHorses = mockData.data.horses;
+  const [allHorses, setAllHorses] = useState([])
 
-  const horses = allHorses.map((horse) => {
-    return (
-      <Link to={`/horses/${horse.id}`} key={horse.id}>
+  const { data } = useQuery(fetchAllHorses, {
+    onCompleted: data => {
+      setAllHorses(() => data.fetchHorses)
+    }})
+
+  return (
```

Comment:
I tend to separate the logic from the return, but it's cool to see it all contained this way. I'm curious if there is a best practice or just preference.